### ISSUE TITLE
Warning causing evaluation to fail

### ISSUE THREAD

**Issue by tgodzik** [scalacenter/scala-debug-adapter#747]
I am getting:
```
compilation failed:
parentheses are required around the parameter of a lambda
Use '-Wconf:msg=lambda-parens:s' to silence this warning. [quickfixable]
Scala 3 migration messages are errors under -Xsource:3. Use -Wconf / @nowarn to filter them or add -Xmigration to demote them to warnings.
Applicable -Wconf / @nowarn filters for this fatal warning: msg=<part of the message>, cat=scala3-migration
```
though the workspace doesn't use -Xsource:3

MAybe we could at least silence the warning?

**Comment by tgodzik** [comment]
I realized I created the issue in Bloop :man_facepalming: 

The issue I had was that expression evaluation was failing in a file which had potential warning when adding -Xsource:3. Do we add that flag by default?

**Comment by adpi2** [comment]
Yes, if it is a Scala 2 entry it seems we had the `-Xsource:3` option.

https://github.com/scalacenter/scala-debug-adapter/blob/50ba5c0cce6364ef2523c8ce5f340d2f86c0b547/modules/core/src/main/scala/ch/epfl/scala/debugadapter/internal/DebugTools.scala#L135

The reason for this is to be able to evaluate expressions in Scala 2 libraries. We don't know what were there compiler options, but it is likely that they use `-Xsource:3` for cross-compilation.

I see 2 possibilities to fix this issue:
1. If it's a module, instead of a library, we don't add any option.
2. Or we remove `-Xfatal-warnings` from the expression compiler options.

I think I prefer solution 2 because, in mixed Scala 2 - Scala 3 projects, we also need to add `-Xtasty-reader` to evaluate Scala 3 symbols in Scala 2 expressions.

**Comment by tgodzik** [comment]
> Or we remove -Xfatal-warnings from the expression compiler options.

Though isn't the warning fatal by default with -Xsource:3 from what I see? https://docs.scala-lang.org/scala3/guides/migration/tooling-scala2-xsource3.html

Maybe we need to change to `-Xsource:3-cross` ?

**Comment by adpi2** [comment]
> Though isn't the warning fatal by default with -Xsource:3 from what I see?

Yes you're right. Then maybe adding `-Xmigration` will help.
